semantic_conventions/http-common.yaml

```diff
@@ -5,10 +5,54 @@ groups:
     prefix: http
     attributes:
       - id: request.method
-        type: string
+        type:
+          allow_custom_values: true
+          members:
+            - id: connect
+              value: "CONNECT"
+              brief: 'CONNECT method.'
+            - id: delete
+              value: "DELETE"
+              brief: 'DELETE method.'
+            - id: get
+              value: "GET"
+              brief: 'GET method.'
+            - id: head
+              value: "HEAD"
+              brief: 'HEAD method.'
+            - id: options
+              value: "OPTIONS"
+              brief: 'OPTIONS method.'
+            - id: patch
+              value: "PATCH"
+              brief: 'PATCH method.'
+            - id: post
+              value: "POST"
+              brief: 'POST method.'
+            - id: put
+              value: "PUT"
+              brief: 'PUT method.'
+            - id: trace
+              value: "TRACE"
+              brief: 'TRACE method.'
+            - id: other
+              value: "other"
```

Comment:
I was originally thinking uppercase `OTHER` to look like the others, but I can see good reasons for lowercase:
* intentionally _doesn't_ look like a real http method
* `other` lowercase could be slightly more reusable(?)

just curious if you had other thoughts on picking lowercase `other`?

Reply:
I suggest that we use something that meets the following conditions:
1. It should be applicable to other semantic conventions.
2. It should be something that users can easily search (e.g. if you ask Google "what does `other` mean for HTTP method", you will find very misleading results).

Reply:
@trask,  `other` came from https://github.com/open-telemetry/opentelemetry-specification/pull/3478#discussion_r1192150868 - the standard enum convention we have on schema uses uppercase.

I agree with @reyang that having a common 'other' (or however we call it) bucket across different attributes would be great.

What do you all think about `_UNKNOWN`. Underscore is a somewhat popular convention for private and obscure things and uppercase creates an impression of a constant rather than a natural language. 

Personally, I'm fine with other variations like `_OTHER`, `UNKOWN`, `_unknown`, and their synonyms.

Reply:
created https://github.com/open-telemetry/semantic-conventions/issues/117

Reply:
I think the underscore is a good idea to highlight this is somewhat "special" and not just a random string. I'm not sure about unknown though, I guess depending how you look at it both Other/unknown have their ups and downs. `_OTHER` looks good to me 👍

Reply:
`_OTHER` seems like a good option to me